lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/MacTrash.java

```diff
@@ -0,0 +1,46 @@
+package org.enso.desktopenvironment;
+
+import java.nio.file.Path;
+import org.graalvm.nativeimage.UnmanagedMemory;
+import org.graalvm.nativeimage.c.function.CFunction;
+import org.graalvm.nativeimage.c.function.CLibrary;
+import org.graalvm.nativeimage.c.type.CCharPointer;
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+import org.graalvm.word.Pointer;
+import org.graalvm.word.WordFactory;
+
+@CLibrary("-framework CoreServices")
+final class MacTrash implements Trash {
+
+  @CFunction
+  static native int FSPathMakeRefWithOptions(
+      CCharPointer path, int flags, Pointer buffer, Pointer any);
+
+  @CFunction
+  static native int FSMoveObjectToTrashSync(Pointer source, Pointer target, int flags);
+
+  @Override
+  public boolean isSupported() {
+    return true;
+  }
+
+  @Override
+  public boolean moveToTrash(Path path) {
+    Pointer ref = WordFactory.nullPointer();
+    Pointer source = UnmanagedMemory.malloc(80);
+    Pointer target = UnmanagedMemory.malloc(80);
```

Comment:
There is a [better version](https://github.com/enso-org/enso/pull/10440#discussion_r1686641212) that conditionally enables itself.